packages/search/addon/services/navi-search-provider.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * This service is used to discover all the available search providers.
+ */
+
+import Service from '@ember/service';
+import { getOwner } from '@ember/application';
+import config from 'ember-get-config';
+
+/* global requirejs */
+
+export default class NaviSearchProviderService extends Service {
+  /**
+   * @method getProvider
+   * @param name
+   * @returns {Object} search provider service object
+   */
+  getProvider(name) {
+    return getOwner(this).lookup(`service:navi-search/${name}`);
+  }
+
+  /**
+   * @method all
+   * @returns {Array} array of available search providers
+   */
+  all() {
+    const searchProvidersRegex = new RegExp(`^(?:${config.modulePrefix}/)?services/navi-search/([a-z-]*)$`),
+      searchProviderServices = Object.keys(requirejs.entries).filter(
+        requirejsFileName =>
+          searchProvidersRegex.test(requirejsFileName) && !requirejsFileName.includes('navi-base-search-provider')
+      );
+    return searchProviderServices.map(providerFileName =>
+      this.getProvider(searchProvidersRegex.exec(providerFileName)[1])
```

Comment:
Why do we need to do the `exec`?  Aren't you already filtering them on line 31?

Reply:
<img width="1307" alt="Screen Shot 2020-02-04 at 10 57 06 AM" src="https://user-images.githubusercontent.com/1649445/73767469-1ccd1180-473d-11ea-874d-4ac1b8cf010c.png">

Reply:
with exec we're pretty much just getting the name of the provider not the whole filepath

Reply:
```js
searchProviderServices = Object.keys(requirejs.entries)
  .filter(requirejsFileName => searchProvidersRegex.test(requirejsFileName))
  .map(fileName => searchProvidersRegex.exec(fileName)[1]));
```